crypto/src/lib.rs

```diff
@@ -69,7 +69,7 @@ impl std::error::Error for NoSuchAlgorithm {}
 
 ffi::ffi_item! {
     /// Algorithm for hashing
-    #[derive(Debug, Clone, Copy, PartialEq, Eq, Display, FfiType)]
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Display, FfiType, IntoSchema, Serialize)]
```

Comment:
Current `Serialize/Deserialize` and `Encode/Decode` of Algorithm is not consistent.

You can check this by running this test: 

```rust

    #[test]
    fn key_pair_serialize_deserialize_consistent() {
        for algorithm in [
            Algorithm::Ed25519,
            Algorithm::Secp256k1,
            Algorithm::BlsNormal,
            Algorithm::BlsSmall,
        ] {
            let key_pair = KeyPair::generate_with_configuration(
                KeyGenConfiguration::default().with_algorithm(algorithm),
            )
            .expect("Failed to generate key pair");
            assert_eq!(
                key_pair,
                serde_json::to_string(&key_pair)
                    .and_then(|key_pair| serde_json::from_str(&key_pair))
                    .expect("Failed to de/serialize key")
            )
        }
    }
``` 

What i would suggest is to add direct implementations of `Deserialize/Serialize` and `Encode/Decode` to the `Algorithm` like that:

```rust
impl Decode for Algorithm {
    // ...
}

impl Encode for Algorithm {
    // ...
}

impl Serialize for Algorithm {
     // ...
}

impl Deserialize for Algorithm {
    // ...
}
```

This way we can simplify implementation of `Deserialize` and `Decode` for `PublicKey/PrivateKey`.

Also we need to add tests to ensure that `Serialize/Deserialize` and `Encode/Decode` of `Algorithm` is indeed consistent (you can use provided test as example).

Reply:
I'd recommend you add this test directly to the crate @Retssaze. Normally we don't need to check that what's serialised is also deserialised accordingly, but in this particular case, given the shift to manually implemented `Serialize`  this is a necessity.